src/android/CameraPreviewFragment.java

```diff
@@ -486,17 +494,28 @@ public void switchCameraTo(String device, CameraSwitchedCallback cameraSwitchedC
                 cameraSwitchedCallback.onSwitch(false);
                 return;
             }
-
-            setUpCamera(device,cameraProvider);
-
+            try {
+                direction = options.getString("direction").equals("front") ? CameraSelector.LENS_FACING_FRONT : CameraSelector.LENS_FACING_BACK;
+            } catch (JSONException e) {
+                direction = CameraSelector.LENS_FACING_BACK;
+            }
+            
+            setUpCamera(options,cameraProvider);
             preview.setSurfaceProvider(viewFinder.getSurfaceProvider());
             cameraSwitchedCallback.onSwitch(true);
         });
     }
     
     @SuppressLint("RestrictedApi")
-    public void setUpCamera(String lens, ProcessCameraProvider cameraProvider) {
+    public void setUpCamera(JSONObject options, ProcessCameraProvider cameraProvider){
         CameraSelector cameraSelector;
+        String lens = null;
```

Comment:
the direction is at class level, this is why we don't handle it here, but it is passed as options  in the switchCameraTo
@HashirRajah @parveshneedhoo your views